MarcusBarnes/mik
# Work on #353: make '[WRITER] preserve_content_filenames' optional in CSVSingleFIle and CSVJson toolchains

**Github issue**: (#353)

# What does this Pull Request do?

Makes the `[WRITER] preserve_content_filenames` setting optional in CSVSingleFIle and CSVJson toolchains.

# What's new?

Assigning a default value of false to this configuration setting now follows the pattern we have established for assigning default .ini values. This change has been added to the CSVSingleFile and CSVJson writers. PHPUnit tests for these toolchains have also been modified to not use the `['WRITER'] 'preserve_content_filenames'` setting.

# How should this be tested?

1. Using configuration files and images in the attached .zip, run MIK with the `['WRITER'] 'preserve_content_filenames'` set to true. Output files should have uppercase filenames (i.e., the names of the original input files).
1. Delete your mik output and temp directories.
1. Rerun MIK but with the `['WRITER'] 'preserve_content_filenames'`commented out. Output files should have lowercase filenames (i.e., the names are based on the CSV record keys).

# Additional Notes
This PR only addresses one part of #353; setting other default values will follow in separate PRs.

This PR will require an update to the relevant toolchain documentation.

# Interested parties
@MarcusBarnes @bondjimbond 

@mark-cooper tagging you also since you worked on the initial implementation of the `[WRITER] preserve_content_filenames` option.

[issue_353.zip](https://github.com/MarcusBarnes/mik/files/876131/issue_353.zip)
